core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -274,14 +277,34 @@ public void setConf(Object newConf) {
   @Override
   public List<TableIdentifier> listTables(SessionContext context, Namespace ns) {
     checkNamespaceIsValid(ns);
+    Map<String, String> queryParams = Maps.newHashMap();
```

Comment:
I feel like this could be consolidated using a `do ... while (...)` since we always need to make a single call.  That might simplify and prevent having two separate call paths.

Reply:
Sure will fix this.